analyzer/engines/magic_bullet_analyzer.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import re
+from spacy.en import English
+NLP = English()
+
+
+
+def enlarge_message(message):
+
+    dummy_context = 'Pretty tinny long short yellow dummy'
+
+    if message.endswith('.'):
+        message = dummy_context + '. ' + message + ' ' + dummy_context
+    else:
+        message = dummy_context + '. ' + message + '. ' + dummy_context
+
+    return message
+
+
+def get_magic_bullet_instance_and_type(pattern):
+
+    result = []
+    magic_bullet_instance = None
+    magic_bullet_type = None
+
+    if '[np = ' in pattern:
+    # e.g. '[np = treatment] as in Dexamethasone treatment'
+        magic_bullet_instance = pattern.replace('[np = ', '').replace(']', '')
+        magic_bullet_type = 'case A'
+    elif '[npl]' in pattern:
+    # e.g. '[npl]is the treatment'
+        magic_bullet_instance = pattern.replace('[npl]', '(\S+ ){4}')
+        magic_bullet_type = 'case B'
+    elif '[npr]' in pattern:
+    # e.g. 'the treatment is[npr]'
+        magic_bullet_instance = pattern.replace('[npr]', '( \S+){4}')
+        magic_bullet_type = 'case C'
+    
+    result.append(magic_bullet_instance)
+    result.append(magic_bullet_type)
+    return result
+
+
+def get_string_match_plus_noun_phrases(magic_bullet_instance, start_word, noun_phrases, message):
+
+    result = [None, noun_phrases]
+    
+    forbidden_tokens = ['the', 'new']
+
+    if magic_bullet_instance in message.lower():
+        if len(noun_phrases) == 0:
+            for np in NLP(message).noun_chunks:
+                np = np.text
+                noun_phrases.append(np)
+        for np in noun_phrases:
+            if magic_bullet_instance in np.lower():
+                if np.split()[-1].lower() == magic_bullet_instance:
+                    if len(np.split()) > 1:
+                        penultimate_token = np.split()[len(np.split())-2].lower()
+                        if penultimate_token not in forbidden_tokens: 
+                            if start_word not in np:
+                                result[0] = np
+                                break
+    
+    return result
+
+
+def get_regex_match(magic_bullet_instance, message):
+
+    result = None
+
+    magic_bullet_instance = re.compile(str(magic_bullet_instance), flags=re.IGNORECASE)
+
+    search_regex = re.search(magic_bullet_instance, message)
+    if search_regex is not None:
+        match = message[search_regex.start():search_regex.end()]
+        result = match
+
+    return result
+
+
+def magic_bullet_analyzer(message, start_word, magic_bullet_grammar, stop_words):
+
```

Comment:
Some comments explaining in plain English what you're doing here would make refactoring much easier.